protos/backend_service.proto

```diff
@@ -0,0 +1,127 @@
+    rpc WaitForOrchestrationCompletion(GetInstanceRequest) returns (GetInstanceResponse);
+
+    // Purges the state of one or more orchestration instances.
+    rpc PurgeOrchestrationState (PurgeInstancesRequest) returns (PurgeInstancesResponse);
+
+    // TODO: Additional operations, such as listing orchestrations
+}
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by the DTFx worker to process orchestration and activity work items.
+service WorkerBackendService {
+    // Starts a bi-directional stream for receiving work items and sending health pings.
+    rpc GetWorkItems(stream GetWorkItemsRequest) returns (stream WorkItem);
+
+    // Gets orchestration runtime state (history, etc.) for a given orchestration instance.
+    rpc GetOrchestrationRuntimeState (GetOrchestrationRuntimeStateRequest) returns (GetOrchestrationRuntimeStateResponse);
+
+    // Completes an outstanding activity work item and adds a new event to the target orchestration's inbox.
+    rpc CompleteActivityWorkItem (CompleteActivityWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Abandons an outstanding activity work item. Abandoned work items will be delivered again after some delay.
+    rpc AbandonActivityWorkItem (AbandonActivityWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Completes an outstanding orchestrator work item, and adds a new event to the target orchestration's inbox.
+    rpc CompleteOrchestrationWorkItem (CompleteOrchestrationWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Abandons an outstanding orchestrator work item. Abandoned work items will be delivered again after some delay.
+    rpc AbandonOrchestrationWorkItem (AbandonOrchestrationWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Fetches metadata about the connected service. This may also be used to verify connectivity.
+    rpc Metadata(google.protobuf.Empty) returns (google.protobuf.Empty); // TODO: Payloads
+}
+
+// Request payload for adding new orchestration events.
+message AddOrchestrationEventRequest {
+    // The ID of the orchestration to send an event to.
+    OrchestrationInstance instance = 1;
+    // The event to send to the orchestration.
+    HistoryEvent event = 2;
+}
+
+// Request parameters for fetching orchestration runtime state.
+message GetOrchestrationRuntimeStateRequest {
+    // The ID of the target orchestration instance.
+    OrchestrationInstance instance = 1;
+}
+
+// Response payload returned when fetching orchestration runtime state.
+message GetOrchestrationRuntimeStateResponse {
+    // The existing history events for the target orchestration instance.
+    repeated HistoryEvent history = 1;
+}
+
+// Request payload for completing an activity work item.
+message CompleteActivityWorkItemRequest {
+    // The completion token that was provided when the work item was fetched.
+    string completionToken = 1;
```

Comment:
```suggestion
    string completion_token = 1;
```

Reply:
Curious as to why Pascal-case is fine for message/service names but snake-case suggested for message fields?

Reply:
Oh, never mind, answered further down.